awx/ui/client/src/notifications/edit/edit.controller.js

```diff
@@ -257,8 +257,13 @@ export default ['Rest', 'Wait',
         $scope.toggleForm = function(key) {
             $scope[key] = !$scope[key];
         };
-        $scope.$watch('notification_type', (value) => {
-            if (value) {
+        $scope.$watch('notification_type', (newValue, oldValue = {}) => {
+            if (newValue) {
+                MessageUtils.updateDefaultsOnScope(
+                  $scope,
+                  defaultMessages[oldValue.value],
+                  defaultMessages[newValue.value]
+                );
```

Comment:
```
else {
    $scope.customize_messages = false;
}
```

Along with a `ngShow: 'notification_type'` added to `customize_messages` over in notificationTemplates.form.js might be nice. Seems to DTRT but no need if it's risky.

Reply:
I just tried this, and it leads to some odd corner cases where, if the user un-chooses a notification type (i.e. selects "Choose a type"), the UI drops any customized messages they've written. I like the idea, but I think this late in the game it has too much potential to introduce some bugs and would make the code more complicated.

I could do only the `ngShow: 'notification_type' change, but that leaves the message fields visible when the toggle switch is hidden.

Reply:
@keithjgrant Sounds good to me. :+1: Thanks for looking into it.